### ISSUE TITLE
[Issue]: renaming a library will trigger a recreation of it and loose all it's settings/adjustments

### ISSUE THREAD

**Issue by da-anda** [jellyfin/jellyfin#8832]
### Please describe your bug

One might think that changing the name of a library is just a cosmetic thing, but it unfortunately isn't and the user is not warned about this fact!

When you change the name, the old library basically will be destroyed and a new one created (or something along this) and you will loose ALL related settings, metadata customization and configuration in all user accounts. So one has to re-enable library access for all user accounts and each user has to reconfigure it's startup page layout and what not. Not to mention the metadata fixes you might have done (not all changes might have been saved in the media folder itself for easy recreation but in a local Jellyfin cache folder instead, which will be killed as well).

### a) Add a text warning to the rename dialog
As a short term "fix", a warning in the rename dialog about these severe implications should be added

### b) Decouple the display name of a library from it's internally used name
Long term, the way libraries are stored and handled needs to be fixed/improved

- use a UUID (or something like that) as internal name, which will be stored in the DB and be used as folder name on the filesystem.
- add an additional `display name` field to the DB (or however this config is stored, not familiar with the codebase) and use it's value in the UI instead of the main identifier (UUID)
- optionial: show the UUID to admins in the library edit dialog so that admins can still figure out the related filesystem folder in case one would have to tinker around in there for whatever reason

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: linux
- Virtualization: Docker
- Clients: Web
- Browser: Firefox
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ristomatti** [comment]
I also got bit by this after tediously manually identifying tens of movies stored on a cloud storage mount.

After playing around with Jellyfin for a while, I realized I can add both my local and remote folders under the same library. Initially I thought the initial scan would be too slow if I wanted to extract thumbnails from the remote files. It turned out to be fast enough to not warrant separate library for local files. 

I then decided to delete my 1 folder local test library "Movies" to be able to free the name for my cloud storage based library I had already updated via manual identification. I figured this way I could avoid identifying them again, but instead lost all the metadata🤦.
